### ISSUE TITLE
Deselect

### ISSUE THREAD

**Issue by serml** [jpnurmi/item_selector#2]
Hi,

I'm trying your library and I find it very useful, thank you very much for your work. I have one question: how can I deselect a selected item? When I tap on a selected item, nothing happens.

Thank you again.

Regards

**Comment by dawit-michael** [comment]
Hi @smunl 
I faced the same issue and this is how i handled it. I hope this could help 
`/// If the old selection is not empty, remove all the items in the old selection and return true. Otherwise, set the old`
`  /// selection to the new selection and return false`
`  /// `
`  /// Args:`
`  ///   start (int): The start index of the selection.`
`  ///   end (int): The end of the selection.`
`  /// `
`  /// Returns:`
`  ///   A boolean value.`
`  bool start(int start, int end) {`
`    if (oldSelection.isNotEmpty) {`
`      removeAll(oldSelection);`
`      oldSelection = ItemSelection();`
`      return true;`
`    } else {`
`      oldSelection = ItemSelection(start, end);`
`      return false;`
`    }`
`  }`

**Comment by jpnurmi** [comment]
@dawit-michael Is this something that could potentially live as a "toggle" selection mode or such in the lib?

**Comment by dawit-michael** [comment]
@jpnurmi It could, potentially , for only single selection. On multiple item selection it will deselect all selected values on single cell click.

**Comment by clragon** [comment]
I was just looking into this package and thats actually very interesting.
so if I have understood this correctly, the package stores the selection as a range, and not as a list of indexes?
that means, you cannot deselect an item that is between other items?

that would actually ruin my potential usecase, where a user is free to pick and choose images from a grid.
drag selection is very common in certain gallery style like UIs, and they allow selection a range of items with it, 
but you are still free to deselect any given item.

**Comment by dawit-michael** [comment]
> I was just looking into this package and thats actually very interesting. so if I have understood this correctly, the package stores the selection as a range, and not as a list of indexes? that means, you cannot deselect an item that is between other items?
> 
> that would actually ruin my potential usecase, where a user is free to pick and choose images from a grid. drag selection is very common in certain gallery style like UIs, and they allow selection a range of items with it, but you are still free to deselect any given item.

You're right. This [package ](https://pub.dev/packages/drag_select_grid_view) might be a good help in your use case . which allows drag select  and  deselect from selection.

**Comment by clragon** [comment]
Ah yes, I actually found this package after finding that one.
Unfortunately, I use specific scroll views (for example staggered_grid_view or grouped_list_view) 
which I cannot replace with a dedicated library gridview.

in my app I have, as of right now, an inherited widget that lives at the top of the route to indicate the selected items to the appbar and items themselves. the selection is actually based on the item data, so its a generic type, and not a list of indexes / index range.

however, my code only allows single tap select and deselect, no complicated mechanisms like drag selecting.
